### ISSUE TITLE
libnetwork plugin ignores ETCD_ENDPOINTS

### ISSUE THREAD

**Issue by fasaxc** [projectcalico/calico-containers#1015]
User **arjun_maya** on slack tried running the following command with `ETCD_ENDPOINTS` set.
```
./calicoctl node --node-image=host1.corpintra.net:5000/caliconode:v0.19 --libnetwork-image=host1.corpintra.net:5000/caliconode-libnetwork:v0.8 --libnetwork
```
But libnetwork failed with the following error when he tried to start a container:
```
Error response from daemon: Cannot start container 8b32db8e1fe969289261061efb658e69019fb6b3129a591ff2d89f9e491b672c: failed to create endpoint desperate_cray on network phpdb: Plugin Error: NetworkDriver.CreateEndpoint, {
 "Err": "Connection to etcd failed due to MaxRetryError(u\"HTTPConnectionPool(host=u'127.0.0.1', port=2379): Max retries exceeded with url: /v2/keys/calico/libnetwork/v1/17472e3a8a2e03e8e4bb556bfb1af5b40b285035b143fa5c771c1ab709f71faa (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f9ad8248090>: Failed to establish a new connection: [Errno 111] Connection refused',))\",)"
}
```
Note: `host=u'127.0.0.1'` suggests it has ignored the `ETCD_ENDPOINTS`.

Setting `ETCD_AUTHORITY` instead made it work.

I suspect we haven't rebuilt the libnetwork plugin since adding `ETCD_ENDPOINTS` to libcalico.

CC @arjunm183

**Comment by ae6rt** [comment]
I ran across this same problem while using this CNI config:

```
$ cat /etc/cni/net.d/10-calico.conf
{
    "name": "calico-k8s-network",
    "type": "calico",
    "etcd_endpoints": "http://10.2.23.113:2379",
    "log_level": "info",
    "ipam": {
        "type": "calico-ipam"
    },
    "kubeconfig": "/etc/kubernetes/kubeconfig"
}
```

**Comment by fasaxc** [comment]
@caseydavenport is the above the correct way to configure the CNI plugin to use etcd_endpoints?

**Comment by ae6rt** [comment]
Not that it's related, but I believe the kubeconfig portion looks like this.  Please correct me if I'm wrong:

```
{
    "name": "calico-k8s-network",
    "type": "calico",
    "etcd_endpoints": "http://10.2.23.113:2379",
    "log_level": "info",
    "ipam": {
        "type": "calico-ipam"
    },
   "kubernetes": {
      "kubeconfig": "/etc/kubernetes/kubeconfig"
   }
}
```

**Comment by caseydavenport** [comment]
@fasaxc yes that looks like the correct CNI network config.  Sounds like this could be a bug in the libcalico used by this CNI plugin version / libnetwork version?

@ae6rt Are you seeing these errors in the CNI log, or when starting `calico/node`?

**Comment by ae6rt** [comment]
@caseydavenport I saw them in the output of _journalctl -u kubelet_.

I'll try to reproduce and send additional information.  I'm very new at this, and need to double check my pronouncements carefully.

**Comment by robbrockbank** [comment]
Fixed in calico-containers 0.21
